File: shared/src/iosMain/kotlin/jp/hotdrop/considercline/IosPlatformDependencies.kt
package jp.hotdrop.considercline

import jp.hotdrop.considercline.db.DriverFactory
import jp.hotdrop.considercline.di.PlatformDependencies
import jp.hotdrop.considercline.repository.local.KmpSharedPreferences

class IosPlatformDependencies: PlatformDependencies {
    override val driverFactory: DriverFactory by lazy { DriverFactory() }
    override val sharedPrefs: KmpSharedPreferences by lazy { KmpSharedPreferences() }
}

File: iosApp/iosApp/IosPlatformDependencies.swift
import shared

class IosPlatformDependencies: PlatformDependencies {
    let driverFactory: DriverFactory
    let sharedPrefs: KmpSharedPreferences
    let httpClientState: HttpClientState

    init() {
        driverFactory = DriverFactory()
        sharedPrefs = IosUserDefaultsStore()
        httpClientState = HttpClientState.useFakeHttp()
    }
}
